hydromt/models/model_mesh.py

```diff
@@ -304,17 +451,69 @@ def write_mesh(self, fn: str = "mesh/mesh.nc", **kwargs) -> None:
             os.makedirs(dirname(_fn))
         self.logger.debug(f"Writing file {fn}")
```

Comment:
Reminder: allow optional_attributes as argument? or use ugriddataset already in self.mesh?